src/coreclr/jit/helperexpansion.cpp

```diff
@@ -764,7 +1146,14 @@ bool Compiler::fgExpandHelperForBlock(BasicBlock** pBlock)
 
         for (GenTree* const tree : stmt->TreeList())
         {
-            if (!tree->IsHelperCall())
+            if (handleIntrinsics)
+            {
+                if (!tree->IsCall() || !(tree->AsCall()->gtCallMoreFlags & GTF_CALL_M_SPECIAL_INTRINSIC))
```

Comment:
```suggestion
                if (!tree->IsCall() || ((tree->AsCall()->gtCallMoreFlags & GTF_CALL_M_SPECIAL_INTRINSIC) == 0))
```

According to our conventions.